networkit/cpp/embedding/BiasedRandomWalk.cpp

```diff
@@ -34,10 +34,28 @@ BiasedRandomWalk::BiasedRandomWalk(const Graph *graph)
     // pre-allocate unordered maps:
     graph->forNodes([&](node v) {
         auto degv = graph->degreeOut(v);
-        for (auto n : graph->neighborRange(v)) {
-            // init index2node:
-            index2node[v].push_back(n);
-            graphData->data[v][n] = AliasSampler(degv);
+        if (graph->isDirected()) {
+            for (auto n : graph->neighborRange(v)) {
+                // local numbering of neighbors by index in index2node[v]:
+                index2node[v].push_back(n);
+            }
+
+            if (!graph->degreeIn(v)) {
+                graphData->data[v][none] = AliasSampler(degv);
+                return;
+            }
+
+            for (auto t : graph->inNeighborRange(v)) {
+                // one AliasSampler with entries for all v-neighbors
+                // per every node t leading to v
+                graphData->data[v][t] = AliasSampler(degv);
+            }
+        } else {
+            for (auto n : graph->neighborRange(v)) {
```

Comment:
You could reserve space also here.
```suggestion
            index2node[v].reserve(graph->degree(v));
            for (auto n : graph->neighborRange(v)) {
```